write_fcdr.f90

```diff
@@ -61,14 +63,23 @@ SUBROUTINE TopLevel()
     LOGICAL :: new_filename
     LOGICAL :: split_single_file
 
-    nArgs = COMMAND_ARGUMENT_COUNT()
-    IF( 14 .gt. nArgs .or. 19 .le. nArgs )THEN
-       CALL Gbcs_Critical(.TRUE.,&
-            'USAGE: ./extract_l1b_data.exe uuid outfile eq_year1 eq_month1 &
-            &eq_day1 eq_hour1 eq_min1 eq_year2 eq_month2 eq_day2 eq_hour2 &
-            &eq_min2 split_single file1 (file2) (file3) (file4) (file5)',&
-            'Main','extract_l1b_data.f90')
-    ENDIF
+!MT: 26-10-2017: Log the list of passed input arguments: 
+!    INTEGER :: i
+!    CHARACTER(LEN=256) :: temp
+!    nArgs = COMMAND_ARGUMENT_COUNT()
+!    print *,'nArgs=',nArgs
+!    DO i = 1,nArgs
+!        CALL GET_COMMAND_ARGUMENT(i,temp,STATUS=stat)
+!        print *,'inArgs=',temp
+!    ENDDO
+
+!    IF( 14 .gt. nArgs .or. 19 .le. nArgs )THEN
```

Comment:
Surely the fix is to either modify this code to deal with more arguments or work out why more arguments are being given and fix if appropriate. All you seem to have done here is stop the error - but not fixed the bug. I think.

Reply:
This is correct. The problem is that some repetition of the orbit file names is being passed which then causes the number of arguments to fall out of the acceptable range 14-19